src/compiler/Stage/InferTypes/AssignIds.elm

```diff
@@ -40,39 +40,52 @@ Output:
 -}
 
 import AST.Canonical as Canonical
+import AST.Common.Located as Located
 import AST.Common.Type as Type
 import AST.Typed as Typed
 import Common
 import Dict.Any
 import Stage.InferTypes.IdSource as IdSource exposing (IdSource)
 
 
-assignIds : Canonical.Expr -> ( Typed.Expr, IdSource )
+assignIds : Canonical.LocatedExpr -> ( Typed.LocatedExpr, IdSource )
 assignIds expr =
     assignIdsWith IdSource.empty expr
 
 
-assignIdsWith : IdSource -> Canonical.Expr -> ( Typed.Expr, IdSource )
+assignIdsWith : IdSource -> Canonical.LocatedExpr -> ( Typed.LocatedExpr, IdSource )
 assignIdsWith idSource expr =
     let
-        wrap : IdSource -> Typed.Expr_ -> ( Typed.Expr, IdSource )
-        wrap idSource_ expr_ =
-            IdSource.one idSource_ (\id -> ( expr_, Type.Var id ))
+        ( expr_, idSource_ ) =
+            assignIdsWithHelp idSource (Located.unwrap expr)
     in
+    {- Keep location, for error context -}
+    ( Located.map (\_ -> expr_) expr
+    , idSource_
+    )
+
+
+assignId : IdSource -> Typed.Expr_ -> ( Typed.Expr, IdSource )
+assignId idSource expr =
+    IdSource.one idSource (\id -> ( expr, Type.Var id ))
+
+
+assignIdsWithHelp : IdSource -> Canonical.Expr -> ( Typed.Expr, IdSource )
+assignIdsWithHelp idSource expr =
```

Comment:
use located like everywhere.

Reply:
This comment seems incorrect? :sweat_smile: